packages/stark-ui/src/modules/route-search/components/route-search.component.ts

```diff
@@ -0,0 +1,187 @@
+	 */
+	public filterRouteEntries(value: string): RouteEntry[] {
+		const filterValue: string = value.toLowerCase();
+		return this.routesToDisplay.filter((state: RouteEntry) => state.label.toLowerCase().indexOf(filterValue) === 0);
+	}
+
+	
+	public ngOnInit(): void {
+		this.pathCtrl.setValue("");
+		this.filteredRouteEntries = this.pathCtrl.valueChanges.pipe(
+			startWith(""),
+			map((option: string) => (option ? this.filterRouteEntries(option) : [...this.routesToDisplay]))
+		);
+		
+		this.hide = true;
+		
+		this.logger.debug(componentName + ": component initialized");
+
+		if (this.routesToDisplay !== undefined){
+			if (this.menuConfig === undefined){
+				this.routesToDisplay =	this.initRoutePathFromRouter();
+			} else{
+				this.routesToDisplay = this.initRoutePathFromConfig();
+			}
+		}
+		this.routesToDisplay = _sortBy(this.routesToDisplay, ['label']);
+	}
+
+	/**
+	 * redirect the user to the chosen path
+	 */
+	public redirect(state: RouteEntry): void {
+		if (state.targetState && state.targetState.url){
+			this.starkRoutingService.navigateTo(state.targetState.url,state.targetStateParams).subscribe(() => {
+				this.pathCtrl.setValue("");
+			});
+		}
+	}
+
+	/**
+	 * Show/hide the input field
+	 */
+	public show(): void {
+		this.hide = !this.hide;
+	}
+	/**
+	 * @ignore
+	 */
+	public trackPath(_index: number, path: RouteEntry): StateDeclaration {
+		if (path.targetState !== undefined){
+			return path.targetState;
+		} else 
+		{
+			return {};
+		}
+	}
+
+	/**
+	 * Retrieve the list of routes from a StarkMenuConfig object which the user have passed to the component
+	 */
+	//tslint:disable:cognitive-complexity
+	public initRoutePathFromConfig(): RouteEntry[]{
+		const routesToDisplay: RouteEntry[] = [];
+		if (this.menuConfig !== undefined && this.menuConfig.menuSections !== undefined) {
+			for (const section of this.menuConfig.menuSections){
+				for (const group of section.menuGroups){
+					if (group.isVisible && group.isEnabled && group.targetState) {
+						const targetState: StateDeclaration = {url: group.targetState, name: group.label, params: group.targetStateParams};
+						routesToDisplay.push({
+							label: group.label,
+							targetState: targetState,
+							targetStateParams: group.targetStateParams
+						});
+					}
+					if (group.entries) {
+						for (const entry of group.entries){
+							if (entry.isEnabled && entry.isVisible) {
+								const targetState: StateDeclaration = {url: entry.targetState, name: entry.label, params: entry.targetStateParams};
+								routesToDisplay.push({
+									label: entry.label,
+									targetState: targetState,
+									targetStateParams: entry.targetStateParams
+								});
+							}
+						}
+					}
+				}
+			}
+		}
+		return routesToDisplay;
+	}
+
+	/**
+	 * Retrieve the list of routes from UIRouter if the user haven't passed a list to the component
+	 */
+	public initRoutePathFromRouter(): RouteEntry[]{
```

Comment:
Can you rename this to `constructRouteEntriesFromRouterStates` so it is more descriptive?